modules/extension/wms/src/main/java/org/geotools/data/ows/Layer.java

```diff
@@ -866,6 +863,7 @@ public GeneralEnvelope getEnvelope(CoordinateReferenceSystem crs) {
                 env = new GeneralEnvelope(new double[] { tempBBox.getMinX(),tempBBox.getMinY() },
                         new double[] { tempBBox.getMaxX(), tempBBox.getMaxY() });
                 env.setCoordinateReferenceSystem(crs);
+                LOGGER.warning("Forcing bbox as " + env);
```

Comment:
The warning is in a catch which is a last resort fallback. 
This fallback is a hack which takes coords from one side, crs from another and puts them together into a GeneralEnvelope; this is very probably going to generate bad data, and really deserves a warning IMHO.